llvm/test/CodeGen/NVPTX/lower-alloca.ll

```diff
@@ -17,7 +17,21 @@ define void @kernel() {
   ret void
 }
 
+define void @alloc_already_in_addrspace5() {
+; LABEL: @lower_alloca_addrspace5
+; PTX-LABEL: .visible .func alloc_already_in_addrspace5(
+  %A = alloca i32, addrspace(5)
```

Comment:
Huh. I don't know that AS-specific alloca is a thing.

Reply:
I didn't knew it was supported, but if it is supported, then NVPTX should be setting this for all `alloca`s.

Reply:
Things in non-generic AS tends to get in the way of optimizations, so doing it up-front will likely result in worse code. Perhaps one day everything will handle non-default AS as well as generic AS, but I don't think we're there yet.

So, for now, I do not see any benefit of allocas in explicit AS -- we can easily infer the AS if/when we need to, and our priority is to eliminate as many of allocas as we can as they are a major bad news for performance, so not getting in the way of LLVM optimizations is more important.

Reply:
Adding this test to make sure the newly added logic is working correctly (instead of asserting). I don't have too much understanding with the optimization tradeoff you mentioned here so I'll defer it to your best judgement. Just wanna make sure if there is any action items I shall take for this PR specifically wrt this test? 🙏

Reply:
The test is OK. Extra comments are helpful, but it's even better if the test is self-documenting. 
E.g. renaming the test above to `alloca_in_explicit_local_as` would do the trick.

Reply:
sounds good, pushed an update.